### PR TITLE
Migrate to OTP2 Trip Props

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
**Description:**

OTP1 trip props are no longer in use: https://github.com/opentripplanner/otp-ui/pull/640.

This PR makes use of the new props and so allows the trip view to work with the graphQL API.

**PR Checklist:**
- [x] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [x] Are all languages supported (Internationalization/Localization)?
- [x] Are appropriate Typescript types implemented?

### CODE REVIEW

**Comment by miles-grant-ibigroup** [comment]
With otp-ui updates all works again @AdrianaCeric

**Comment by AdrianaCeric** [comment]
Ok let me try again

**Review by AdrianaCeric — APPROVED** [review]
Looks good! (Tested on ST DA)

File: lib/components/viewers/trip-viewer.js [hunk 1]
```diff
@@ -88,6 +88,13 @@ class TripViewer extends Component {
       timeZone: homeTimezone
     })
 
+    const fromIndex = tripData?.stops?.findIndex(
+      (stop) => stop.stopId === viewedTrip?.fromStopId
+    )
+    const toIndex = tripData?.stops?.findIndex(
+      (stop) => stop.stopId === viewedTrip?.toStopId
+    )
+
```
Comment (amy-corson-ibigroup) [inline lib/components/viewers/trip-viewer.js:97]: Could pull this out into its own function like `findStationIndex` and then pass the `fromStopId` or `toStopId` respectively? Up to you!

**Review by amy-corson-ibigroup — APPROVED** [review]
We are so back

File: lib/components/viewers/trip-viewer.js [hunk 2]
```diff
@@ -133,20 +140,16 @@ class TripViewer extends Component {
               screenreader or keyboard nav, the departure, arrival, and stop viewer links 
               are all accessible without having to go through all the stops not on the trip. */}
 
-              {viewedTrip?.fromIndex && (
+              {fromIndex && (
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply lib/components/viewers/trip-viewer.js:143]: Wow amazing catch thank you!

File: lib/components/viewers/trip-viewer.js (hunk 1 above)
Comment (miles-grant-ibigroup) [reply lib/components/viewers/trip-viewer.js:97]: I think that would wind up being more code overall... Especially since this view is on its way out anyway I think this can be a future improvement
